pkg/controller/remotemachineset/awsactuator_test.go

```diff
@@ -158,7 +159,9 @@ func TestAWSActuator(t *testing.T) {
 				}(),
 			},
 			mockAWSClient: func(client *mockaws.MockClient) {
-				mockDescribeSubnets(client, []string{"zone1", "zone1", "zone2"}, []string{"subnet-zone1", "subnet-zone2", "subnet-zone3"})
+				mockDescribeSubnets(client, []string{"zone1", "zone1", "zone2"},
+					[]string{"subnet-zone1", "subnet-zone2", "subnet-zone3"}, "vpc-1")
+			// What's a neat way of creating a mock client with both DescribeSubnets and DescribeRouteTablesPages here?
```

Comment:
If I'm understanding the issue, it should simple be creating a mockRouteTables function that accepts the client reference as well, and calling it here exactly as you do for mockDescribeSubnets. This mockAWSClient is a function reference the test will call to build the mock client, but you can have additional statements to configure that client as you wish. See dnszone_controller_test.go for an example.

Reply:
Yes - writing up the mock function is my real problem :/
https://github.com/openshift/hive/pull/1234/files#diff-3686c203cc4ff7e788327e6a9261edccd61dd438114c0e389a975d567f760143R426-R439

Reply:
Apologies I missed that other comment.

One thought, do you need the Pages variant or would DescribeRouteTables be ok?

Or, perhaps even better, we do have one other client call using pages, and it is mocked here: https://github.com/openshift/hive/blob/de7dd1d6a81e0c48360aa4e52bb29c64d1d9f8a0/pkg/controller/dnszone/awsactuator_test.go#L178

Do either of those help?

Reply:
Hmmm I suppose DescribeRouteTables should be okay. Installer uses the Pages variant for some reason even though the code they imported from upstream Kube uses the simpler describe (https://github.com/kubernetes/kubernetes/blob/9f036cd43d35a9c41d7ac4ca82398a6d0bef957b/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L3302)
I'm guessing the pages variant will be better in the case where there are many routetables? Perhaps I shouldn't be needlessly complicating things... Thanks!